Bypass cache for XSRF Tokens

HermesSbicego-Laser opened the pull request:
solves #8469

HermesSbicego-Laser committed:
Replace RequestVerificationTokenWithBeaconTag in order to support cached pages that contain forms.

HermesSbicego-Laser committed:
Enforce the condition for which we need to prevent caching the RequestVerificationToken

sebastienros commented on src/Orchard.Web/Modules/Orchard.OutputCache/Filters/OutputCacheFilter.cs:
```diff
@@ -588,6 +608,44 @@ public class OutputCacheFilter : FilterProvider, IActionFilter, IResultFilter, I
+                string patternRegion = "<\\s*input\\s*.*name\\s*=\\s*\"__RequestVerificationToken\"\\s*.*value\\s*=\\s*\"" +
+                                       "(?<value>[\\w-]*)" +
+                                       "\"\\s*/>";
+                RegexOptions regexOptions = RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.Compiled;
```
Try not to use a regex, it's dangerous. Or at least define a small timeout value. Also before using the regex, so a simple IndexOf to ensure there is a potential match first.

HermesSbicego-Laser replied:
The new commit adds indexof check and a regex timeout as suggested. I modified the regex replace with a static one in order to be more performant. 
In our dev environment:
With the regex object it performs a replace in 20.68 millisenconds
With the static regex it performs a replace in 4.23 millisenconds

sebastienros replied:
But do you really need a regex? Why not simply search for `<input name="__RequestVerificationToken"` since it's generated by some well-known code. ? If you search for spaces between `name` and `=` why not seach for other attributes before `name`? Because you are making an assumption about what is generating this input, and the same assumption lets you search for the exact string I suggested, removing the need of a regex.

HermesSbicego-Laser replied:
I removed regex as suggested, see latest commit.

HermesSbicego-Laser committed:
Upgrades YamlDotNet from 9.1.3 to 11.1.1 because 9.1.3 is not more available as nuget package

HermesSbicego-Laser committed:
Merge branch 'fix/8471_upgrade_YamlDotNet' into fix/8469_bypass_cache_XSRF_Token

HermesSbicego-Laser committed:
- Adds IndexOf check in order to run Regex.Replace only if we assume there is at least a match
- Sets Regex.Replace timeout to 2000 milliseconds
- Uses static Regex.Replace in order to gain in performances

HermesSbicego-Laser committed:
removed log info

HermesSbicego-Laser committed:
- removes regex replace with string replace. It searches for <input name="__RequestVerificationToken" since it's generated by some well-known code.
